lib/ldp.js

```diff
@@ -432,19 +432,20 @@ class LDP {
     }
   }
 
-  getAvailablePath (host, containerURI, { slug = uuid.v1(), extension }) {
-    const path = slug + extension
+  async getAvailableUrl (hostname, containerURI, { slug = uuid.v1(), extension }) {
+    const fileName = slug + extension
```

Comment:
This is going to give encoding problems.

Here's what we will need to do. We need to do all operations fully in URL space (expect for the one "does the file exist already?" check, for which we convert to file space).

1. Construct the full URL of the target resource.
2. Map it to a filename with `mapUrlToFile`.
3. Check if the file exists.
4. If it exists, generate a new full URL (with `uuid`)
5. Return the URL.